sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1081,7 +1081,7 @@ case class ArrayContains(left: Expression, right: Expression)
     (left.dataType, right.dataType) match {
       case (_, NullType) => Seq.empty
       case (ArrayType(e1, hasNull), e2) =>
-        TypeCoercion.findTightestCommonType(e1, e2) match {
+        TypeCoercion.findWiderTypeForTwo(e1, e2) match {
```

Comment:
This kind of things need a justification and migration guide update with tests. Which JIRA caused this issue?

Reply:
PR https://github.com/apache/spark/pull/22408 which is to prevent implicit downcasting of right expression.
It doesn't handle the case of decimal type upcasting.

Reply:
I will update migration guide.